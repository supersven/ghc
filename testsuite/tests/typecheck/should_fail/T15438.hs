{-# LANGUAGE  MultiParamTypeClasses, RankNTypes #-}

module T15438 where

class C a b

foo :: (forall a b. C a b => b -> b) -> Int
foo = error "urk"
