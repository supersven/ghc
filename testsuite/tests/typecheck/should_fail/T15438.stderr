
T15438.hs:7:8: error:
    • Could not deduce (C a0 b)
      from the context: C a b
        bound by a type expected by the context:
                   forall a b. C a b => b -> b
        at T15438.hs:7:8-43
      The type variable ‘a0’ is ambiguous
    • In the ambiguity check for ‘foo’
      To defer the ambiguity check to use sites, enable AllowAmbiguousTypes
      In the type signature: foo :: (forall a b. C a b => b -> b) -> Int
